.github/workflows/publish.yaml

```diff
@@ -78,7 +78,7 @@ jobs:
           echo "Tagging with ${{ inputs.image-tag }}"
           echo ${{ secrets.GITHUB_TOKEN }} | docker login $IMAGE_REGISTRY -u $GITHUB_ACTOR --password-stdin
           docker pull $API_IMAGE
-          printf "$(git log -n 1 --format=format:'%h% %>(12) %cs %aN%  %s %d')" > ./api/src/version.txt
+          printf "$(git log -n 1 --format=format:'%h% %>(12) %cs %aN%  "%s" %d')" > ./api/src/version.txt
```

Comment:
Have you tried `git log -n 1 --format=format:'%d' --decorate-refs=refs/tags`. Then you would only get the tag